Unity-Technologies/Graphics
# PBR Sky renderer now shares its internal precomputation table between different instances.

### Purpose of this PR
Currently the Physically Based Sky Renderer has precomputation tables that are not shared between cameras using the same sky. This leads to a big overhead of GPU memory. In order to fix that, the PBR sky renderer now caches these tables internally to be shared between instances of the renderer (one renderer per camera).

Also:
- Removed duplicated hash code probably due to a bad merge.
- Added a debug visualizer for HDCamera object.

---
### Testing status

**Manual Tests**: What did you do?
- Tested a complex scene with one PBR sky and multiple realtime probes and cameras. Tested via the debugger and memory profiler that only one set of precomputed tables were allocated.
- Tested the same scene with multiple different PBR sky and checked that the right number of precomputed tables were allocated.
- Tested that making a new set or setting the sky to none properly deallocated the tables.
@Unity-Technologies/gfx-qa-hdrp those tests + automated tests should be enough.
